Add Slick FK error handler.

houcros opened the pull request.

koshelev commented on libats-slick/src/main/scala/com/advancedtelematic/libats/slick/db/SlickExtensions.scala:
```diff
@@ -29,6 +29,15 @@ object SlickPipeToUnit {
 trait SlickResultExtensions {
   implicit class DBIOActionExtensions[T](action: DBIO[T]) {
+
+    def handleForeignKeyError(error: Throwable)(implicit ec: ExecutionContext): DBIO[T] = {
```
Why is not it a part of `handleIntegrityErrors`?

houcros replied:
With the new FK `campaigns -> updates` I thought it would be nice to have a different error message when there is a FK integrity error, instead of the generic one we have now with `handleIntegrityErrors`.
I mean here: https://github.com/advancedtelematic/campaigner/blob/d68340eabab51eb086a51a639ca1fefd0ba00023/src/main/scala/com/advancedtelematic/campaigner/db/Repository.scala#L216

koshelev commented:
Added a commit, WDYT about doing it this way?

houcros commented:
Nice, that gives us way more flexibility 👍

simao commented on libats-slick/src/test/scala/com/advancedtelematic/libats/slick/db/SlickExtensionsSpec.scala:
```diff
@@ -104,7 +104,7 @@ class SlickExtensionsSpec extends FunSuite with Matchers with ScalaFutures with
     val error = new RuntimeException("[test] expected error") with NoStackTrace
     val g = BookMeta(1, 1984, 0)
-    val f = db.run((bookMeta += g).handleForeignKeyError(error))
+    val f = db.run((bookMeta += g).mapError { case SqlExceptions.NoReferencedRow(_) => error })
```
would still be nice to have a shorthand for `handleForeignKeyError`, which could be written in terms of `mapError`

houcros commented:
I rewrote `handleForeignKeyError` and `handleIntegrityErrors` in terms of `mapError`.
Also I made some changes to `mapError` because otherwise the error was being thrown even when there was no FK violation. Have a look and correct me if I'm wrong, I added a test to check for that.

houcros commented:
Amended last commit @koshelev
